### DISCUSSION TITLE
Plugins with secret variables in configuration files

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#325] [Product Feedback]
### Which section?

Marketplace

### Which is the version?

-

### Description

Hi all!

Some plugins (e.g. files-service) requires to write some secret variables within configuration files. In the context of Kubernetes, this lead a conscientious user to have the whole configuration file as a secret, even though only a field of it is actually secret.

It would be desirable to have, for these plugins, a feature that allows the secret fields of a configuration file to be interpolated with the environment variables of the plugin. In such a way I can keep my configuration file as a configmap on Kubernetes and  only the secret field as a secret.

What do you guys think?
Thanks.

**Comment by fredmaggiowski** [comment]
Hi, we are aware of this issue but it struggles to end up in our roadmap as it leads to several possible breaking changes in lots of components, if this becomes a blocking issue we can understand the priorities and collaborate with the different component maintainers to solve the problem

**Reply by lucascanna** [reply]
Hi @fredmaggiowski, 

Thanks for the reply. I am happy to hear you are aware of the issue.
This was just a feedback and, at the moment, does not represent a blocking issue for us.
